From: Miaohe Lin <linmiaohe@huawei.com>
Subject: mm: remove unneeded PageAnon check in restore_exclusive_pte()
Date: Sat, 16 Jul 2022 16:18:16 +0800

When code reaches here, the page must be !PageAnon.  There's no need to
check PageAnon again.  Remove it.

Link: https://lkml.kernel.org/r/20220716081816.10752-1-linmiaohe@huawei.com
Signed-off-by: Miaohe Lin <linmiaohe@huawei.com>
Reviewed-by: David Hildenbrand <david@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memory.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/memory.c~mm-remove-unneeded-pageanon-check-in-restore_exclusive_pte
+++ a/mm/memory.c
@@ -753,7 +753,7 @@ static void restore_exclusive_pte(struct
 		 * Currently device exclusive access only supports anonymous
 		 * memory so the entry shouldn't point to a filebacked page.
 		 */
-		WARN_ON_ONCE(!PageAnon(page));
+		WARN_ON_ONCE(1);
 
 	set_pte_at(vma->vm_mm, address, ptep, pte);
 
_
